src/est/import/lib/EFI/Import/Source/Accession.pm

```diff
@@ -0,0 +1,158 @@
+
+package EFI::Import::Source::Accession;
+
+use strict;
+use warnings;
+
+use Data::Dumper;
+
+use Cwd qw(abs_path);
+use File::Basename qw(dirname);
+use lib dirname(abs_path(__FILE__)) . "/../../../"; # Import libs
+use lib dirname(abs_path(__FILE__)) . "/../../../../../../../lib"; # Global libs
+use parent qw(EFI::Import::Source);
+
+use EFI::Import::Metadata ':source';
+
+use EFI::Util::FASTA::Headers;
+
+
+our $TYPE_NAME = "accessions";
+
+
+sub new {
+    my $class = shift;
+    my %args = @_;
+
+    my $self = $class->SUPER::new(%args);
+    $self->{_type} = $TYPE_NAME;
+    $self->{use_headers} = 1;
+
+    return $self;
+}
+
+
+sub init {
+    my $self = shift;
+    my $config = shift;
+    my $efiDb = shift;
+    $self->SUPER::init($config, $efiDb, @_);
+
+    my $file = $config->getConfigValue("accessions");
+    $self->{acc_file} = $file;
+    $self->{efi_db} = $efiDb // die "Require efi db argument";
+
+    if (not $self->{acc_file}) {
+        $self->addError("Require --accessions arg");
+        return undef;
+    }
+
+    return 1;
+}
+
+
+# Returns a list of sequence IDs that are in the specified families (provided via command-line argument)
+sub getSequenceIds {
+    my $self = shift;
+
+    my $rawIds = $self->parseAccessions();
+    my ($ids, $metadata) = $self->identifyAccessionIds($rawIds);
+
+    $self->addSunburstIds($ids);
+
+    #TODO: add sequences from family
+    #TODO: apply tax/family filters here??? ???
+    my $numRemoved = 0;
+
+    $self->addStatsValue("num_filter_removed", $numRemoved);
+
+    my $seqType = $self->{uniref_version} ? $self->{uniref_version} : "uniprot";
+    return {ids => $ids, type => $seqType, meta => $metadata};
+}
+
+
+####################################################################################################
+# 
+#
+
+
+sub parseAccessions {
+    my $self = shift;
+
+    $self->{logger}->message("Parsing accession file $self->{acc_file}");
```

Comment:
This line (and a similar call on line 120) cause a crash:
```
Using accessions as source
Retrieving accession IDs from source efi_db.sqlite
Can't call method "message" on an undefined value at src/est/import/lib/EFI/Import/Source/../../../EFI/Import/Source/Accession.pm line 82.
```
Looks like the logger isn't initialized?

Reply:
Both of those logger statements can/should be removed.